monitoring/grafana/KubeStellarControllersMonitoring.json

```diff
@@ -0,0 +1,760 @@
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
+  "editable": true,
+  "fiscalYearStartMonth": 0,
+  "graphTooltip": 0,
+  "id": 7,
+  "links": [],
+  "panels": [
+    {
+      "datasource": {
+        "default": true,
+        "type": "prometheus",
+        "uid": "prometheus-test"
+      },
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "palette-classic"
+          },
+          "custom": {
+            "axisBorderShow": false,
+            "axisCenteredZero": false,
+            "axisColorMode": "text",
+            "axisLabel": "",
+            "axisPlacement": "auto",
+            "barAlignment": 0,
+            "barWidthFactor": 0.6,
+            "drawStyle": "line",
+            "fillOpacity": 0,
+            "gradientMode": "none",
+            "hideFrom": {
+              "legend": false,
+              "tooltip": false,
+              "viz": false
+            },
+            "insertNulls": false,
+            "lineInterpolation": "linear",
+            "lineWidth": 1,
+            "pointSize": 5,
+            "scaleDistribution": {
+              "type": "linear"
+            },
+            "showPoints": "auto",
+            "spanNulls": false,
+            "stacking": {
+              "group": "A",
+              "mode": "none"
+            },
+            "thresholdsStyle": {
+              "mode": "off"
+            }
+          },
+          "decimals": 2,
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
+              }
+            ]
+          },
+          "unit": "s"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 0
+      },
+      "id": 1,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
+        },
+        "tooltip": {
+          "mode": "single",
+          "sort": "none"
+        }
+      },
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "disableTextWrap": false,
+          "editorMode": "builder",
+          "expr": "histogram_quantile(0.99, sum by(instance, name, le) (rate(workqueue_queue_duration_seconds_bucket{namespace=\"$SpaceNS\", job=\"$KSController\"}[5m])))",
+          "fullMetaSearch": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A",
+          "useBackend": false
+        }
+      ],
+      "title": "99th Percentile of Sync Duration for Items in the Workqueue ",
```

Comment:
This distribution deserves more exposition. Here is what I see in my idling system after `test/e2d/run-test.sh`:

<img width="436" alt="Screenshot 2024-08-12 at 2 30 45 PM" src="https://github.com/user-attachments/assets/683ff457-6856-4b73-baad-59e98dae6703">

10 seconds is quite surprising and alarming. But then one has to remember that PromQL histogram quantiles have more precision than is justified by the data, which are quantized to the available buckets. Sadly, the viewer of such a display does not see what the buckets are. Perhaps that would be worth putting in the "description" of the panel?

For comparison, here is the average (but without the title adjusted to say that it is the average):

<img width="474" alt="Screenshot 2024-08-12 at 2 30 36 PM" src="https://github.com/user-attachments/assets/34dd11e7-0cff-4f83-93d7-ce13cf7a1bc0">

Here is the median:

<img width="476" alt="Screenshot 2024-08-12 at 2 31 10 PM" src="https://github.com/user-attachments/assets/1d9ef59e-7b0c-470f-8f3f-b5c9149bc693">

Here is the 75th percentile:

<img width="481" alt="Screenshot 2024-08-12 at 2 31 41 PM" src="https://github.com/user-attachments/assets/c17341f6-45c3-419b-8558-784cab1b3fa2">

Here is the 25th percentile:

<img width="458" alt="Screenshot 2024-08-12 at 2 31 57 PM" src="https://github.com/user-attachments/assets/1e4dd854-cbe1-4cc0-9be2-f218d2413c5b">

You could even put multiple traces on one graph.

Reply:
Fixed. I added traces for the 75%, 50% and 25%. I couldn't find a way to add a information of the percentile in the legend of the graph. Hence, I added it in the description.